astropy/io/fits/src/compressionmodule.c

```diff
@@ -877,9 +958,7 @@ PyObject* compression_compress_hdu(PyObject* self, PyObject* args)
     FITSfile* Fptr = NULL;
     int status = 0;
 
-    if (!PyArg_ParseTuple(args, "O:compression.compress_hdu", &hdu))
-    {
-        PyErr_SetString(PyExc_TypeError, "Couldn't parse arguments");
```

Comment:
If this call fails it will already set an exception so no need to overwrite that exception.